network/src/p2p.rs

```diff
@@ -296,3 +317,43 @@ impl ReliableNetwork<WorkerMessage> for P2pNetwork {
         self.send(peer, f).await
     }
 }
+
+#[async_trait]
+impl UnreliableNetwork<WorkerBatchRequest> for P2pNetwork {
+    type Response = WorkerBatchResponse;
+    async fn unreliable_send(
+        &mut self,
+        peer: NetworkPublicKey,
+        message: &WorkerBatchRequest,
+    ) -> JoinHandle<anyhow::Result<anemo::Response<WorkerBatchResponse>>> {
+        let message = message.to_owned();
+        let f = move |peer| async move {
+            WorkerToWorkerClient::new(peer)
+                .request_batches(message)
+                .await
+        };
+        self.unreliable_send(peer, f).await
+    }
+}
+
+#[async_trait]
+impl ReliableNetwork<WorkerBatchRequest> for P2pNetwork {
```

Comment:
Not for this PR, but the two definitions of unreliable and reliable networks feels like a bit of boilerplate. Is it possible to avoid defining the `send()` and `unreliable_send()` functions for other types of request and response?

And just curious, what is the difference in implementation between unreliable and reliable networks?

Reply:
I agree these are starting to look repetitive enough that a macro or something might be worthwhile (if it's not possible to do via a default impl in the trait itself). +1 for a future PR.

Essentially the reliable variant does a bunch of retrying in case of errors or missing peer connections, while unreliable does not.

Reply:
Traits have default methods: we can request the implementer to do less in the required method of the traits, e.g.implement the part where they return a `BoxFuture`.